### ISSUE TITLE
CA2119 not triggered when extended interface is implemented

### ISSUE THREAD

**Issue by jskeet** [dotnet/roslyn-analyzers#4406]
### Analyzer

**Diagnostic ID**: [CA2119](https://docs.microsoft.com/dotnet/fundamentals/code-analysis/quality-rules/ca2119): `Seal methods that satisfy private interfaces`

### Analyzer source

**NuGet Package**: [Microsoft.CodeAnalysis.FxCopAnalyzers](https://www.nuget.org/packages/Microsoft.CodeAnalysis.FxCopAnalyzers)

**Version**: v3.3.1 (Latest)

### Describe the bug

When there are two internal interfaces, one extending another, methods from the "super" interface aren't spotted by CA2119 when a class implements the "extended" interface

### Steps To Reproduce

Source code:

```csharp
namespace FxCopRule
{
    internal interface IInternal1
    {
        void Method();
    }

    internal interface IInternal2 : IInternal1
    {
    }

    public abstract class ImplementationBase : IInternal2
    {
        public abstract void Method();
    }
}
```

### Expected behavior

CA2119 should be raised, as `ImplementationBase` still implements `IInternal1.Method` with an abstract method.

### Actual behavior

CA2119 is not raised.

## Additional context

Changing the class declaration to explicitly implement the "super" interface does cause the warning to be raised:

```csharp
public abstract class ImplementationBase : IInternal1, IInternal2
```

**Comment by Evangelink** [comment]
@jskeet Do you expect to have `IInternal2` implementing `IInternal1`?

**Comment by jskeet** [comment]
@Evangelink: Whoops, yes, I did! I'll double check that I tested it correctly, although I believe I did. I'll add a comment when I've retested, although it might not be today. Huge apologies for messing that up, and thanks for pointing it out.

**Comment by Evangelink** [comment]
I will make a test on master and will post the results here.

**Comment by jskeet** [comment]
Right, now retested - and yes, it's still a valid bug report :) I've edited the first comment.

**Comment by Evangelink** [comment]
I fixed the issue in between ;)
